tests/test_mytardis_client_rest_factory.py

```diff
@@ -70,5 +77,113 @@ def test_backoff_on_mytardis_rest_factory(
+    datafile_get_response_single: dict[str, Any],
+    auth: AuthConfig,
+    connection: ConnectionConfig,
+) -> None:
+    mt_client = MyTardisRESTFactory(auth, connection)
+
+    responses.add(
+        responses.GET,
+        mt_client.compose_url("/dataset_file"),
+        status=200,
+        json=datafile_get_response_single,
+    )
+
+    datafiles, meta = mt_client.get("/dataset_file", object_type=Datafile)
+
+    assert isinstance(datafiles, list)
+    assert len(datafiles) == 1
+    assert datafiles[0].resource_uri == URI("/api/v1/dataset_file/0/")
+    assert datafiles[0].obj.filename == "test_filename.txt"
+
+    assert isinstance(meta, GetResponseMeta)
+    assert meta.total_count == 1
+
+
+@responses.activate
+def test_mytardis_client_rest_get_multi(
+    datafile_get_response_multi: dict[str, Any],
+    auth: AuthConfig,
+    connection: ConnectionConfig,
+) -> None:
+    mt_client = MyTardisRESTFactory(auth, connection)
+
+    responses.add(
+        responses.GET,
+        mt_client.compose_url("/dataset_file"),
+        status=200,
+        json=datafile_get_response_multi,
+    )
+
+    datafiles, meta = mt_client.get(
+        "/dataset_file",
+        object_type=Datafile,
+        meta_params=GetRequestMetaParams(offset=2, limit=3),
+    )
+
+    assert isinstance(datafiles, list)
+    assert len(datafiles) == 30
+    assert (isinstance(df, Datafile) for df in datafiles)
+
+    assert datafiles[0].resource_uri == URI("/api/v1/dataset_file/0/")
+    assert datafiles[1].resource_uri == URI("/api/v1/dataset_file/1/")
+    assert datafiles[2].resource_uri == URI("/api/v1/dataset_file/2/")
+
+    assert datafiles[0].obj.filename == "test_filename_0.txt"
+    assert datafiles[1].obj.filename == "test_filename_1.txt"
+    assert datafiles[2].obj.filename == "test_filename_2.txt"
+
+    assert isinstance(meta, GetResponseMeta)
+    assert meta.total_count == 30
+    assert meta.limit == 30
+    assert meta.offset == 0
+
+
+@responses.activate
+def test_mytardis_client_rest_get_all(
+    datafile_get_response_paginated_first: dict[str, Any],
+    datafile_get_response_paginated_second: dict[str, Any],
+    auth: AuthConfig,
+    connection: ConnectionConfig,
+) -> None:
+    mt_client = MyTardisRESTFactory(auth, connection)
+
+    responses.add(
+        responses.GET,
+        mt_client.compose_url("/dataset_file"),
+        status=200,
+        json=datafile_get_response_paginated_first,
+    )
+    responses.add(
+        responses.GET,
+        mt_client.compose_url("/dataset_file"),
+        status=200,
+        json=datafile_get_response_paginated_second,
+    )
+
+    datafiles, total_count = mt_client.get_all(
+        "/dataset_file",
+        object_type=Datafile,
+        batch_size=20,
+    )
+
+    assert isinstance(datafiles, list)
+    assert len(datafiles) == 30
+    assert (isinstance(df, Datafile) for df in datafiles)
+
+    assert datafiles[0].resource_uri == URI("/api/v1/dataset_file/0/")
```

Comment:
could do the assert check in a loop here to check all 30 give you're confirming more than one already.

Reply:
Good point, I've added that in. I was a bit reluctant as sometimes if you loop things and get an error, the message can be less explicit, so it's harder to figure out the issue. Will see how it goes though. In the C++ gtest framework, there's some nice stuff for testing that sequences of possibly different types are equal in some sense, I wonder if there's anything similar for python